src/NewTwitchApi/Resources/EventSubApi.php

```diff
@@ -0,0 +1,286 @@
+     */
+    public function subscribeToChannelHypeTrainBegin(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'begin', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelhype_trainprogress
+     */
+    public function subscribeToChannelHypeTrainProgress(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'progress', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelhype_trainend
+     */
+    public function subscribeToChannelHypeTrainEnd(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'end', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#stream-subscriptions
+     */
+    private function subscribeToStream(string $twitchId, string $event, string $callback, string $bearer, string $secret): void
+    {
+        if (!in_array($event, ['online', 'offline'])) {
+            throw new \InvalidArgumentException('Invalid value for stream event type. Accepted values: online,offline.');
+        }
+
+        $this->subscribe(
+            sprintf('stream.%s', $event),
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#streamonline
+     */
+    public function subscribeToStreamOnline(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToStream($twitchId, 'online', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#streamoffline
+     */
+    public function subscribeToStreamOffline(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToStream($twitchId, 'offline', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#userauthorizationrevoke
+     */
+    public function subscribeToUserAuthorizationRevoke(string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribe(
+            'user.authorization.revoke',
+            '1',
+            ['client_id' => $this->clientId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#userupdate
+     */
+    public function subscribeToUserUpdate(string $twitchId, string $callback, string $bearer, string $secret): void
```

Comment:
- Bearer should always be the first parameter
- Since Secrets are required for all Subscriptions, put this directly after the Bearer
- Since Callbacks are required for all Subscriptions, put this after the Secret